### PR TITLE
create new repo using github api

### PR DESCRIPTION
**Pull request by miltontony**

### CODE REVIEW

File: project/templates/unicoremc/new_project.html [hunk 1]
```diff
@@ -181,12 +181,25 @@
...
+        var app_name = $("#unicore_app").val();
+        var country = $("#country").val().toLowerCase();
+        var data = {
+            "name": "unicore-cms-content-" + app_name + "-" + country + "-test",
+            "description": "A Unicore CMS content repo for "+ app_name + " " + country,
+            "homepage": "https://github.com",
+            "private": false,
+            "has_issues": true
+        }
+        return $.post("https://api.github.com/miltontony/repos", data)
```
Comment (smn) [inline project/templates/unicoremc/new_project.html]: How is auth handled here?

File: unicoremc/tests/test_states.py [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+
+        try:
+            shutil.rmtree(
+                os.path.join(settings.CMS_REPO_PATH, 'test-source-repo'))
+        except:
+            pass
+
+        workdir = os.path.join(settings.CMS_REPO_PATH, 'test-source-repo')
+        os.makedirs(workdir)
+        self.source_repo = Repo.init(workdir)
```
Comment (smn) [inline unicoremc/tests/test_states.py:32]: Perhaps use elasticgit and use [TestCase.addCleanup](https://docs.python.org/2/library/unittest.html#unittest.TestCase.addCleanup)?

File: project/templates/unicoremc/new_project.html (hunk 1 above)
Comment (smn) [inline project/templates/unicoremc/new_project.html]: How is this different from what's happening here: https://github.com/miltontony/unicore-mc/pull/5/files#diff-cfbd53eb7022e8dc9b36819d7bd13e17R38?

**Comment by smn** [comment]
:+1:

File: project/templates/unicoremc/new_project.html (hunk 1 above)
Comment (miltontony) [inline project/templates/unicoremc/new_project.html]: @smn I need to delete this stuff - it's before I started on the StateMachine implementation

File: unicoremc/tests/test_states.py (hunk 2 above)
Comment (miltontony) [inline unicoremc/tests/test_states.py:32]: @I'll look at this in the next PR

**Pull request merged**